File: Programmers/pro_12953_N개의 최소공배수.py
def gcd(a,b):
    while(b):
        r=a%b
        a=b
        b=r
    return a
def solution(arr):
    for i in range(len(arr)):
        arr[i]=(arr[i]*arr[i-1])//gcd(arr[i],arr[i-1])
    return arr[-1]

File: Programmers/pro_12953_N개의 최소공배수.js
const gcd = (a, b) => {
    for (let r = 0; b; r = a % b, a = b, b = r);
    return a
}

function solution(arr) {
    return Array(arr.length - 1).fill(0).reduce((acc, _, i) => {
        const [a, b] = [acc, arr[i + 1]]
        return acc = a / gcd(a, b) * b
    }, arr[0])
}